### PR TITLE
Add exclusions to implementation dependencies in gradle.build for Emulator.

### PR DESCRIPTION
**Pull request by alex-iotiq**
Partial solution for this issue https://github.com/jenkinsci/peass-ci-plugin/issues/171

### CODE REVIEW

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java [hunk 1]
```diff
@@ -102,19 +102,32 @@ private void editGradlefileContents(final File tempFolder, final GradleBuildfile
...
+               dependencyGradle = String.join("\n", excludes);
             } else {
                dependencyGradle = "implementation '" + dependency.getGradleDependency() + "'";
             }
+
             visitor.addLine(visitor.getDependencyLine() - 1, dependencyGradle);
          }
          if (testTransformer.getConfig().getExecutionConfig().isUseAnbox()) {
-            visitor.addLine(visitor.getDependencyLine() - 1, "   androidTestImplementation 'androidx.test:rules:1.4.0'");
+            visitor.addLine(visitor.getDependencyLine() - 1, "    androidTestImplementation 'androidx.test:rules:1.4.0'");
```
Comment (DaGeRe) [inline dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:130]: Why is this change needed?

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java [hunk 2]
```diff
@@ -102,19 +102,32 @@ private void editGradlefileContents(final File tempFolder, final GradleBuildfile
    private void addDependencies(final GradleBuildfileVisitor visitor) {
       JUnitVersions versions = testTransformer.getJUnitVersions();
       boolean isExcludeLog4j = testTransformer.getConfig().getExecutionConfig().isExcludeLog4jSlf4jImpl();
+      boolean isAnbox = testTransformer.getConfig().getExecutionConfig().isUseAnbox();
+      final String peass = " // Added dynamically by Peass.";
```
Comment (DaGeRe) [inline dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:106]: What does the peass-string to? As far as I see it, it is never used again.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java [hunk 3]
```diff
@@ -102,19 +102,32 @@ private void editGradlefileContents(final File tempFolder, final GradleBuildfile
...
       if (visitor.getDependencyLine() != -1) {
          for (RequiredDependency dependency : RequiredDependency.getAll(versions)) {
             final String dependencyGradle;
             if (isExcludeLog4j && dependency.getMavenDependency().getArtifactId().contains("kopeme")) {
                String excludeString = "{ exclude group: '" + MavenPomUtil.LOG4J_GROUPID + "', module: '" + MavenPomUtil.LOG4J_SLF4J_IMPL_ARTIFACTID + "' }";
                dependencyGradle = "implementation ('" + dependency.getGradleDependency() + "') " + excludeString;
+            } else if (isAnbox && dependency.getGradleDependency().contains("kopeme")) {
```
Comment (DaGeRe) [inline dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:113]: Is this really the intended if statement? Because this will only happen if log4j is not excluded - will this never happen, or is it just for current implementation convenience, that you do not handle the case of `isAnbox && isExcludeLog4j`? If the latter is true, I'd recommend to add a small comment that this case should be addressed in the future.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java [hunk 4]
```diff
@@ -102,19 +102,32 @@ private void editGradlefileContents(final File tempFolder, final GradleBuildfile
...
+                  "        exclude group: '"+ "org.hamcrest" + "', module: '" + "hamcrest'" + peass, 
+                  "        exclude group: '"+ "org.aspectj" + "', module: '" + "aspectjrt'" + peass, 
+                  "        exclude group: '"+ "org.apache.logging.log4j" + "', module: '" + "log4j-core'" + peass, 
+                  "    }" + peass,
+               };
+               dependencyGradle = String.join("\n", excludes);
             } else {
                dependencyGradle = "implementation '" + dependency.getGradleDependency() + "'";
             }
+
```
Comment (DaGeRe) [inline dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:126]: Please do not change the formatting if not explicitely required / makes sense for some reason (it just adds noise to a diff and makes it harder to read). In this case, I do not see the value of this empty line.

**Review by DaGeRe — CHANGES_REQUESTED** [review]
Thanks for the PR! This basically looks fine, but I see some discussion points. Let's discuss this in our meeting later on.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java (hunk 1 above)
Comment (alex-iotiq) [reply dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:130]: To match the indentation to the other dependencies.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java (hunk 2 above)
Comment (alex-iotiq) [reply dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:106]: It's used in lines 115-120 as comments to show that the added changes to the build.gradle file were automatically added.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java (hunk 4 above)
Comment (alex-iotiq) [reply dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:126]: I didn't intend to add this line. I'll remove it.

File: dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java (hunk 3 above)
Comment (alex-iotiq) [reply dependency/src/main/java/de/dagere/peass/execution/gradle/GradleBuildfileEditor.java:115]: I've added a comment for the future.

**Comment by DaGeRe** [comment]
Looks fine, thanks for the PR!
